Examples/Nodify.Playground/BaseSettingViewModel.cs

```diff
@@ -0,0 +1,31 @@
+﻿namespace Nodify.Playground
+{
+    public class BaseSettingViewModel<T> : ObservableObject, ISettingViewModel
+    {
+        public string Name { get; }
+        public string? Description { get; } // This is the tooltip
+
+        private object? _value;
+
+        object? ISettingViewModel.Value
+        {
+            get => _value;
+            set => SetProperty(ref _value, value);
+        }
+
+        public SettingsType Type { get; set; }
+
+        public T Value
+        {
+            get => ((ISettingViewModel)this).Value is T val ? val : default!;
+            set => ((ISettingViewModel)this).Value = value;
+        }
+
+        public BaseSettingViewModel(string name, SettingsType type, string? description = default)
```

Comment:
`type` parameter can be removed and `Type` property can be calculated inside the constructor based on type of T. (e.g. switch expression on typeof(T) to map the type to a SettingsType)